server/issue.go

```diff
@@ -324,6 +330,24 @@ func (s *Server) composeIssueRelationship(ctx context.Context, issue *api.Issue)
 	return nil
 }
 
+// Only allow Owner/DBA as the assignee, not Developer.
+func (s *Server) validateAssigneeRoleByID(ctx context.Context, assigneeID int) error {
+	assignee, err := s.PrincipalService.FindPrincipal(ctx, &api.PrincipalFind{
+		ID: &assigneeID,
+	})
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to find principal by id %d", assigneeID)).SetInternal(err)
+	}
+	if assignee == nil {
+		return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Principal ID not found: %d", assigneeID))
+	}
+	if assignee.Role == api.Developer {
```

Comment:
use assignee.Role != api.OWNER ||  assignee.Role != api.DBA

Otherwise, adding a new role and forget changing here will break this logic

Reply:
Updated with `assignee.Role != api.OWNER && assignee.Role != api.DBA`.